src/components/Button/Button.scss

```diff
@@ -119,11 +119,9 @@ $stacking-order: (
 .Icon {
   transition: color duration() easing();
 
-  &:first-child {
-    // This compensates for the typical icon used in buttons being
-    // inset within its bounding box.
-    margin-left: -(spacing(extra-tight));
-  }
+  // This compensates for the typical icon used in buttons being
+  // inset within its bounding box.
+  margin-left: -(spacing(extra-tight));
```

Comment:
I couldn't find a good reason to limit this to the first child. When there are multiple icons in a button it is either a generic icon, the loading spinner, or the disclosure. As shown in my tophatting gif, these 3 cases are accounted for (the last button in the gif is a good example of a button using all 3).